src/main/java/meditracker/medication/MedicationManager.java

```diff
@@ -72,6 +77,30 @@ private static void checkForDuplicateMedication(String name) throws MediTrackerE
         }
     }
 
+    /**
+     * Checks if the user has input a valid expiry date format.
+     * Also checks that the expiry date entered is not expired.
+     *
+     * @param expiryDate Expiry Date in yyyy-MM-dd format
+     * @throws DateTimeParseException When the date entered is in the wrong format
+     * @throws MediTrackerException When the date entered is already expired
+     */
+    public static void checkforValidExpiryDate(String expiryDate)
```

Comment:
Might be better to have this method in `Medication` instead as it does not have to deal with the context of all `Medication` but just itself only

Reply:
Implemented in [de50cef](https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/220/commits/de50cefd7e288993053326b5d43ea2fce6ef35d3)